### DISCUSSION TITLE
选型 LuaJIT 相较 JS V8引擎，有哪些优劣？

### DISCUSSION THREAD

**Discussion by shar0** [ejoy/ant#66] [General]
现代计算设备普遍CPU性能过剩是可见的。
但依然好奇为什么重新设计游戏引擎，仍选择了Lua而非JS/TS，或是不像Godot一样设计自己的语言？

**Comment by cloudwu** [comment]
我们喜欢 Lua 。另外，我不喜欢 LuaJIT 也不喜欢 javascript 。

我也更倾向于用 C/Lua 解决性能问题，而不是 JIT / AOT 技术。前者更有确定性。专注于动态语言和 C/C++ 打交道，Lua 是目前最佳选择。

设计语言相对于实现一门 xxx like 的语言是一件非常困难的工作。一不小心，就会设计成 javascript 那样的问题语言，需要不断的补丁去填坑，并留下无数历史问题。我特别欣赏 Lua 开发团队在语言设计上的谨慎态度。

**Reply by heshan0x1** [reply]
我们项目是 mmo 类型，也是属于 lua 重度使用者，对 lua 版本的选择一直很别扭，想换 5.4，但是性能上看和 luajit 确实还是有一定的差距，直接限制了大副本的玩家人数，只能捏着鼻子用 beta 版本的 luajit

**Reply by cloudwu** [reply]
如果试都没试过的话，那说明”性能“只是个借口，真实原因是懒得换。

不然，就应该是给出真实的比较数据了，而不是”臆想“ 限制了玩家人数。

**Comment by actboy168** [comment]
ant使用的是lua5.4而非luajit。v8的二进制就比整个ant的所有二进制加起来要大几倍，属实有点多余。

**Comment by vkensou** [comment]
因为云风喜欢lua

**Comment by najoast** [comment]
JavaScript 的流行，可以说是人类的悲哀。在历史的关键时刻，把一门如此不堪的语言推到了这个位置，再随着 Web 的爆炸式发展而发展，逐渐成为一个所有人都不得不接受的事实，不管它的设计有多糟糕。
